### PR TITLE
Migrate classes and tests for truss element from solid mechanics

### LINKED ISSUE #6789: Migrating classes for truss element from solid mechanics module to tensor mechanics module
### Description of the enhancement or error report
The classes with tests in solid mechanics module for truss element problems are now migrating to tensor mechanics module

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

### PR DESCRIPTION
**Pull request by hchen139**
Closes #6789

### CODE REVIEW

File: modules/tensor_mechanics/src/kernels/StressDivergenceTensorsTruss.C [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  else
+  {
+    unsigned int coupled_component = 0;
+    bool active = false;
+
+    for (unsigned int i = 0; i < _ndisp; ++i)
+      if (jvar == _disp_var[i])
+      {
+        coupled_component = i;
+        active = true;
```
Comment (dschwen) [inline modules/tensor_mechanics/src/kernels/StressDivergenceTensorsTruss.C]: you can add a `break;` here as further iterations of the loop are pointless

File: modules/tensor_mechanics/src/kernels/StressDivergenceTensorsTruss.C [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+void
+StressDivergenceTensorsTruss::computeOffDiagJacobian(unsigned int jvar)
+{
+  if (jvar == _var.number())
+    computeJacobian();
+  else
+  {
+    unsigned int coupled_component = 0;
+    bool active = false;
```
Comment (dschwen) [inline modules/tensor_mechanics/src/kernels/StressDivergenceTensorsTruss.C]: active is a weird name. Myebe you can make it more descriptive. Especially below, where `active == false` contains the logic that handles the coupling to other variables.

**Comment by dschwen** [comment]
If you push further changes I suggest putting the issue number in teh commit message

**Comment by dschwen** [comment]
Looks good otherwise. 👍

**Comment by permcody** [comment]
👍

**Pull request merged**